packages/teuchos/parameterlist/src/Teuchos_StandardParameterEntryValidators.hpp

```diff
@@ -715,20 +715,25 @@ class TEUCHOSPARAMETERLIST_LIB_DLL_EXPORT AnyNumberParameterEntryValidator
   //@{
 
   /** \brief Determines what type is the preferred type. */
-  enum EPreferredType { PREFER_INT, PREFER_DOUBLE, PREFER_STRING };
+  enum EPreferredType { PREFER_INT, PREFER_LONG_LONG, PREFER_DOUBLE, PREFER_STRING };
 
 
   /** \brief Determines the types that are accepted.  */
   class AcceptedTypes {
   public:
     /** \brief Allow all types or not on construction. */
     AcceptedTypes( bool allowAllTypesByDefault = true )
```

Comment:
`AcceptedTypes` reminds me of the C++ Executors proposal, with preferred / required properties for executors.